com.unity.render-pipelines.core/Editor/ContextualMenuDispatcher.cs

```diff
@@ -54,99 +54,91 @@ static void RemoveCameraComponent(MenuCommand command)
+            if (typeof(T).GetCustomAttributes(typeof(RequireComponent), true).FirstOrDefault() is RequireComponent rc)
             {
-                if (additionalData.GetCustomAttributes(typeof(RequireComponent), true).FirstOrDefault() is RequireComponent rc)
+                using (ListPool<Type>.Get(out var componentsToRemove))
                 {
-                    string types = rc.m_Type0.Name;
+                    componentsToRemove.Add(rc.m_Type0);
                     if (rc.m_Type1 != null)
-                        types += $", {rc.m_Type1.Name}";
+                        componentsToRemove.Add(rc.m_Type1);
                     if (rc.m_Type2 != null)
-                        types += $", {rc.m_Type2.Name}";
+                        componentsToRemove.Add(rc.m_Type2);
 
-                    MenuManager.AddMenuItem($"CONTEXT/{additionalData.Name}/Remove Component",
-                        string.Empty,
-                        false,
-                        0,
-                        () => EditorUtility.DisplayDialog($"Remove {additionalData.Name} is blocked", $"You can not delete this component, you will have to remove the {types}.", "OK"),
-                        () => true);
+                    if (EditorUtility.DisplayDialog(
+                        title: $"Are you sure you want to proceed?",
+                        message: $"This operation will also remove {string.Join($"{Environment.NewLine} - ", componentsToRemove)}.",
+                        ok: $"Remove everything",
+                        cancel: "Cancel"))
+                    {
+                        T additionalData = command.context as T;
+                        foreach (var type in componentsToRemove)
+                        {
+                            RemoveComponent(type, additionalData.GetComponent(type));
+                        }
+                    }
                 }
             }
+            else
+                throw new Exception($"Missing attribute {typeof(RequireComponent).FullName} on type {typeof(T).FullName}");
         }
 
         static void RemoveComponent<T>(MenuCommand command)
             where T : Component
         {
             T comp = command.context as T;
-
-            if (!DispatchRemoveComponent<T>(comp))
-            {
-                //preserve built-in behavior
-                if (RemoveComponentUtils.CanRemoveComponent(comp, RemoveComponentUtils.ComponentDependencies(comp)))
-                    Undo.DestroyObjectImmediate(command.context);
-            }
+            RemoveComponent(typeof(T), comp);
         }
 
-        static bool DispatchRemoveComponent<T>(T component)
-            where T : Component
+        /// <summary>
+        /// Removes the component and it's additional datas
+        /// </summary>
+        /// <param name="type">Type to remove</param>
+        /// <param name="comp">The component</param>
+        public static void RemoveComponent(Type type, Component comp)
```

Comment:
I don't see any occurence in user code, can we make this private?

Reply:
It was being used on test code, added internal classes to test the feature